django/db/models/query.py

```diff
@@ -818,12 +818,26 @@ def _combinator_query(self, combinator, *other_qs, all=False):
         return clone
 
     def union(self, *other_qs, all=False):
+        # If the query is an EmptyQuerySet, start from the with first nonempty
+        # query from other_qs.
+        if isinstance(self, EmptyQuerySet):
+            qs = [qs for qs in other_qs if not isinstance(qs, EmptyQuerySet)]
+            return qs[0]._combinator_query('union', *qs[1:], all=all) if len(qs) > 0 else self
```

Comment:
An alternative to duplicating the `_combinator_query('union'` stuff could be:
```
if not qs:
    return qs
self, other_qs = qs[0], qs[1:]
```
I'm not completely in love with that. If you prefer the current version, at least change `if len(qs) > 0` to `if qs` as there's no need to count the length, correct?